src/main/java/org/candlepin/hostedtest/HostedTestCloudRegistrationAdapter.java

```diff
@@ -190,6 +192,27 @@ public CloudAccountData setupCloudAccountOrg(String cloudAccountID, String cloud
         return new CloudAccountData("owner_key", false);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String checkCloudAccountOrgIsReady(String cloudAccountID, CloudProvider cloudProviderShortName,
+        String cloudOfferingID)
+        throws OrgForCloudAccountNotCreatedYetException, OrgForCloudAccountNotEntitledYetException {
+        String ownerKey = this.datastore.getOwnerKeyForCloudAccountId(cloudAccountID);
+
+        if (ownerKey == null || ownerKey.isBlank()) {
+            throw new OrgForCloudAccountNotCreatedYetException();
+        }
+
+        Set<String> productIdsForOfferId = this.datastore.getProductIdsForOfferId(cloudOfferingID);
+        if (!isOwnerEntitledToProducts(ownerKey, productIdsForOfferId)) {
+            throw new OrgForCloudAccountNotEntitledYetException();
+        }
+
+        return this.datastore.getOwnerKeyForCloudAccountId(cloudAccountID);
```

Comment:
No need to call this method again, you already fetched the ownerKey at the start of the method

Reply:
done